tap/tcp_reader.go

```diff
@@ -101,6 +108,9 @@ func (reader *tcpReader) Read(p []byte) (int, error) {
 	}
 
 	l := copy(p, reader.data)
+	if !reader.isProtocolIdentified() {
+		reader.buffer = append(reader.buffer, reader.data...)
+	}
```

Comment:
This could duplicate data on `reader.buffer`.
```go
reader.data = reader.buffer
reader.buffer = append(reader.buffer, reader.data...)
```
is the same as
```
reader.buffer = append(reader.buffer, reader.buffer...)
```

Only run this when getting new messages (`msg.GetBytes()`).

Reply:
@nimrod-up9 fixed with https://github.com/up9inc/mizu/commit/ce840940b3bd1012f431233742fc0c73c3f0afa7